File: bank-service/src/main/java/rs/edu/raf/dto/CreditTypeCreateDTO.java
package rs.edu.raf.dto;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import jakarta.validation.constraints.PositiveOrZero;

import java.math.BigDecimal;

public record CreditTypeCreateDTO(@NotNull String name, @NotNull @Positive BigDecimal nominalInterestRate,
                                  @PositiveOrZero Long currencyId, int minLoanTerm, int maxLoanTerm,
                                  @NotNull @Positive BigDecimal maxLoanAmount, BigDecimal prepayment) {
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/dto/CreditTypeCreateDTO.kt
package org.example.bankservicekotlin.dto

import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Positive
import jakarta.validation.constraints.PositiveOrZero
import java.math.BigDecimal

data class CreditTypeCreateDTO(
    @NotNull val name: String,
    @NotNull @Positive val nominalInterestRate: BigDecimal,
    @PositiveOrZero val currencyId: Long,
    val minLoanTerm: Int,
    val maxLoanTerm: Int,
    @NotNull @Positive val maxLoanAmount: BigDecimal,
    val prepayment: BigDecimal
)
